components/scream/src/share/tests/common_physics_functions_tests.cpp

```diff
@@ -317,6 +349,9 @@ void run(std::mt19937_64& engine)
 
     // Check vmr(mmr(vmr))==mmr (up to roundoff tolerance)
     REQUIRE ( Check::approx_equal(mmr_host(k),mmr_for_testing_host(k),test_tol) );
+
+    // Check drymmr==mmr (up to roundoff tolerance)
+    REQUIRE ( Check::approx_equal(drymmr_host(k),mmr_for_testing_host(k),test_tol) );
```

Comment:
@singhbalwinder , yes, these changes look better.  Although I still think it would be good to have the mmr variables in the wet/dry mmr tests be completely independent from the mmr/vmr variables.  I would suggest making a `drymmr_for_testing` variable, like what is done with `mmr_for_testing` and then testing against that here.  If you do that, you will need to use `drymmr_for_testing` in line 300 above.  That way you are testing that if you start with `drymmr_for_testing` and then convert to `wetmmr` and then back to `drymmr` the `drymmr_for_testing` and `drymmr` match.

Reply:
You will also want to follow the example on line 141 to generate random values for `drymmr_for_testing`.

Also, a total side note, but the mmr/vmr conversions actually assume that mmr is wet given that this will be the scream default.   So the use of the `mmr` variable as "dry" could cause confusion.  But here, if we make the variables independent we won't have to worry about that.

Reply:
Thanks for the very thoughtful suggestions. I have created variables which are completely independent for these functions. I have also changed the tests so that they assume wet mixing ratios to start with. I went through the code again but let me know if I need to change anything or if there are any typos/misleading comments.